server/graphql/common/expenses.ts

```diff
@@ -382,14 +382,38 @@ export const canApprove: ExpensePermissionEvaluator = async (req, expense, optio
     return false;
   } else {
     expense.collective = expense.collective || (await req.loaders.Collective.byId.load(expense.CollectiveId));
-    if (hasPolicy(expense.collective, POLICIES.EXPENSE_AUTHOR_CANNOT_APPROVE) && req.remoteUser.id === expense.UserId) {
-      if (options?.throw) {
-        throw new Forbidden(
-          'User cannot approve their own expenses',
-          EXPENSE_PERMISSION_ERROR_CODES.AUTHOR_CANNOT_APPROVE,
-        );
+    expense.collective.host =
+      expense.collective.host || (await req.loaders.Collective.byId.load(expense.collective.HostCollectiveId));
+
+    const hostPolicy = getPolicy(expense.collective.host, POLICIES.EXPENSE_AUTHOR_CANNOT_APPROVE);
+    const collectivePolicy = getPolicy(expense.collective, POLICIES.EXPENSE_AUTHOR_CANNOT_APPROVE);
+
+    const policy = hostPolicy.enabled && hostPolicy.appliesToHostedCollectives ? hostPolicy : collectivePolicy;
+    const currency = hostPolicy.enabled ? expense.collective.host.currency : expense.collective.currency;
+
+    if (policy.enabled && expense.amount >= policy.amountInCents) {
+      const collectiveAdminCount = await req.loaders.Member.countAdminMembersOfCollective.load(expense.collective.id);
+
+      const shouldEnforcePolicy =
+        collectiveAdminCount >= 2 ||
+        (collectivePolicy.enabled && !hostPolicy.enabled) ||
```

Comment:
Having this inside a conditional for `policy.enabled` after evaluating that from `collectivePolicy` and `hostPolicy` looks a bit convoluted. Could you add a comment on what exactly we're testing here?

```ts
        (collectivePolicy.enabled && !hostPolicy.enabled) ||
```

Reply:
I agree 100%, I tried to refactor it a bit now. let me know what you think.